### PR TITLE
Handle RTL in modals

### LINKED ISSUE #2759: RTL - Modals
### Expected Behavior

All the modals that use `ReactModal` directly (not the Modal component) should work in RTL:
- [ ] browser modal
- [ ] import modal
- [ ] preview modal
- [ ] webgl modal

Splitting out from #1579

### PR DESCRIPTION
**Pull request by chrisgarrity**
### Resolves

_What Github issue does this resolve (please include link)?_

- Resolves #2759 

### Proposed Changes

_Describe what this Pull Request does_
Wraps the modal content in a `<div dir=...` and sets the direction based on the `isRtl` state. Handles all the modals that use `ReactModal` directly not the `Modal` component.

### Test Coverage
Try it:
https://chrisgarrity.github.io/scratch-gui/feature/2759-modals/?locale=he

Manual Testing:
- [ ] Beta modal - button layout is rtl
- [ ] import project modal - back arrow is reversed and input field is reversed
- [ ] WebGL modal - text is RTL
- [ ] unsupported browser modal - text is RTL

### Browser Coverage
Check the OS/browser combinations tested (At least 2)

Mac
 * [x] Chrome 
 * [ ] Firefox 
 * [ ] Safari
 * [x] Opera - gets unsupported browser modal
 
Windows
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Edge
 
Chromebook
 * [ ] Chrome
 
iPad
* [ ] Safari

Android Tablet
* [ ] Chrome

### CODE REVIEW

File: src/containers/error-boundary.jsx [hunk 1]
```diff
@@ -57,15 +58,25 @@ class ErrorBoundary extends React.Component {
...
+    isRtl: PropTypes.bool
 };
 
-export default ErrorBoundary;
+const mapStateToProps = state => ({
+    isRtl: state.locales.isRtl
+});
+
+export default connect(
+    mapStateToProps
```
Comment (kchadha) [inline src/containers/error-boundary.jsx]: Does there need to be an empty `mapDispatchToProps` here?

Comment (chrisgarrity) [reply src/containers/error-boundary.jsx]: Hmmm, it's an optional parameter so I didn't think it needed it. However, re-reading the docs it says that it injects `dispatch` into your props if you don't provide one. Looks like we should provide an empty one if we don't want the `dispatch prop` warning.

**Comment by chrisgarrity** [comment]
I didn't reverse the browser image because it's just a facsimile of a browser window, but I just checked on one of the tablets and the arrows are to the right of the address bar in Chrome so perhaps it should. The other parts didn't change because everything is centered.

File: src/containers/error-boundary.jsx (hunk 1 above)
Comment (chrisgarrity) [reply src/containers/error-boundary.jsx]: Or maybe it's fine... backpack container doesn't define `mapDispatchToProps`: https://github.com/LLK/scratch-gui/blob/6a9bb4089c3d965c99f9b46e7b830f15ce750276/src/containers/backpack.jsx#L156

**Comment by chrisgarrity** [comment]
@kchadha I've added the no-op mapDispatchToProps to the errorBoundary container.

**Comment by fsih** [comment]
Not a big deal but, I think both arguments to connect are optional and not passing mapDispatchToProps is very common

C:\Users\DD\Git\scratch-gui\src\components\gui\gui.jsx:
  318  });
  319  
  320: export default injectIntl(connect(
  321:     mapStateToProps
  322: )(GUIComponent));
  323  

C:\Users\DD\Git\scratch-gui\src\components\stage-header\stage-header.jsx:
  180  };
  181  
  182: export default injectIntl(connect(
  183:     mapStateToProps
  184: )(StageHeaderComponent));
  185  

C:\Users\DD\Git\scratch-gui\src\containers\custom-procedures.jsx:
  191  });
  192  
  193: export default connect(
  194:     mapStateToProps
  195: )(CustomProcedures);
  196  

C:\Users\DD\Git\scratch-gui\src\containers\modal.jsx:
   58  });
   59  
   60: export default connect(
   61:     mapStateToProps
   62: )(Modal);
   63  

C:\Users\DD\Git\scratch-gui\src\containers\paint-editor-wrapper.jsx:
   79  };
   80  
   81: export default connect(
   82:     mapStateToProps
   83: )(PaintEditorWrapper);
   84  

C:\Users\DD\Git\scratch-gui\src\containers\sound-editor.jsx:
  226  };
  227  
  228: export default connect(
  229:     mapStateToProps
  230: )(SoundEditor);

**Comment by kchadha** [comment]
@fsih then are we wrong about what creates this warning? Because it wasn't always there...

image

**Comment by fsih** [comment]
@kchadha That happens because we have a tendency to use "...props" to pass props to components and then accidentally leave too many props in there. Div isn't a react component, so it doesn't know what to do with dispatch. This change gets rid of the warning for me: https://github.com/LLK/scratch-gui/commit/eb0383ba59cd828c27f4a447d3fd75eaa17b883b

**Pull request merged**